search_indices/views/base.py

```diff
@@ -68,10 +69,23 @@ class BaseSearchDocumentViewSet(BaseDocumentViewSet):
         "_score",
     )
 
-    def filter_queryset(self, queryset):
+    def _filter_search_fields_for_unauthenticated(self):
+        search_fields = []
+        for field in self.search_fields:
+            if "InformationSystem" in field:
+                continue
+            search_fields.append(field)
+        self.search_fields = tuple(search_fields)
+
+        for attribute in attributes_for_authenticated:
+            self.filter_fields.pop(attribute, None)
+
+    def initial(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return super().initial(request, *args, **kwargs)
+
         # Restrict querying information system queries to authenticated users.
         # The information system field contents are not public.
-        if not self.request.user.is_authenticated:
-            self.filter_fields.pop("record_InformationSystem", None)
+        self._filter_search_fields_for_unauthenticated()
 
-        return super().filter_queryset(queryset)
+        super().initial(request, *args, **kwargs)
```

Comment:
Not sure if this method should return the value of the super call or not, but it shouldn't do both of those. Now it does that for authenticated users and doesn't do it for unauthenticated users, which seems odd.

Reply:
Ah, its a mistake. The initial method is not expected to return anything - I'll change this a bit